sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -482,6 +482,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
               errorClass = "UNBOUND_SQL_PARAMETER",
               messageParameters = Map("name" -> p.name))
 
+          case ma@MultiAlias(child, names) if child.resolved && !child.isInstanceOf[Generator] =>
```

Comment:
Why exactly do we check for `child.resolved` here?

Reply:
If child is unresolved, we cannot check for Generator, as we would usually have UnresolvedFunction.